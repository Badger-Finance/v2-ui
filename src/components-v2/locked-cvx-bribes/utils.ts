import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
