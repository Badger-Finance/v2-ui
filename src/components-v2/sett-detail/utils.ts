import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
