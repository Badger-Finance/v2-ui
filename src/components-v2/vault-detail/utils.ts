import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
