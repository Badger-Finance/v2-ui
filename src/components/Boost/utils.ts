import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
