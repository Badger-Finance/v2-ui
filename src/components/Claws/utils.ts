import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
