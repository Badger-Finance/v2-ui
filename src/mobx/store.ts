import { RouterStore } from 'mobx-router';
import UiState from './reducers';
import WalletStore from './stores/walletStore';
import ContractsStore from './stores/contractsStore';
import AirdropStore from './stores/airdropStore';
import RebaseStore from './stores/rebaseStore';
import RewardsStore from './stores/rewardsStore';
import IbBTCStore from './stores/ibBTCStore';
import TransactionsStore from './stores/transactionsStore';
import SettStoreV2 from './stores/settStoreV2';
import { NETWORK_LIST } from '../config/constants';
import HoneyPotStore from './stores/honeyPotStore';

export class RootStore {
	public router: RouterStore<RootStore>;
	public wallet: WalletStore;
	public uiState: UiState;
	public contracts: ContractsStore;
	public airdrops: AirdropStore;
	public rebase: RebaseStore;
	public rewards: RewardsStore;
	public ibBTCStore: IbBTCStore;
	public setts: SettStoreV2;
	public transactions: TransactionsStore;
	public honeyPot: HoneyPotStore;

	constructor() {
		this.router = new RouterStore<RootStore>(this);
		this.wallet = new WalletStore(this);
		this.contracts = new ContractsStore(this);
		this.airdrops = new AirdropStore(this);
		this.rebase = new RebaseStore(this);
		this.rewards = new RewardsStore(this);
		this.uiState = new UiState(this);
		this.ibBTCStore = new IbBTCStore(this);
		this.honeyPot = new HoneyPotStore(this);
		// RenVM transactions store.
		this.transactions = new TransactionsStore(this);
		this.setts = new SettStoreV2(this);

		this.walletRefresh();
	}

	async walletRefresh(): Promise<void> {
		const chain = this.wallet.network.name;
		const refreshData = [
			this.setts.loadAssets(chain),
			this.setts.loadPrices(chain),
			this.wallet.getGasPrice(),
			this.contracts.updateProvider(),
		];
		if (chain === NETWORK_LIST.ETH) {
			refreshData.push(this.setts.loadGeysers(chain));
			refreshData.push(this.rebase.fetchRebaseStats());
		} else {
			refreshData.push(this.setts.loadSetts(chain));
		}
		await Promise.all(refreshData);

		if (this.wallet.connectedAddress) {
			this.contracts.updateProvider();
			await this.wallet.getGasPrice();
			await this.contracts.fetchContracts();
			if (chain === NETWORK_LIST.ETH) {
				this.uiState.reduceRebase();
				this.ibBTCStore.init();
				this.rewards.fetchSettRewards();
				this.uiState.reduceTreeRewards();
				this.airdrops.fetchAirdrops();
				this.uiState.reduceAirdrops();
			}
			this.uiState.reduceStats();
		}
	}
}

const store = new RootStore();

export default store;
