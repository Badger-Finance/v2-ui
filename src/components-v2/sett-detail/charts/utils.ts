import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
