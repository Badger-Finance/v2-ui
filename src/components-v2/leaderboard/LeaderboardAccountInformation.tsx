import React, { useContext } from 'react';
import { Button, Grid, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { observer } from 'mobx-react-lite';
import { StoreContext } from '../../mobx/store-context';
import { formatWithoutExtraZeros, numberWithCommas } from '../../mobx/utils/helpers';
import { useConnectWallet } from '../../mobx/utils/hooks';
import routes from '../../config/routes';
import { Skeleton } from '@material-ui/lab';
import { getRankNumberFromBoost } from '../../utils/componentHelpers';
import { LEADERBOARD_RANKS } from '../../config/constants';

const useStyles = makeStyles((theme) => ({
	amountToNextRank: {
		fontSize: 12,
	},
	justifyCenterOnMobile: {
		[theme.breakpoints.down('xs')]: {
			justifyContent: 'center',
		},
	},
	boostContainer: {
		[theme.breakpoints.down('xs')]: {
			textAlign: 'center',
			marginBottom: 16,
		},
	},
	calculateBoostContainer: {
		[theme.breakpoints.down('xs')]: {
			marginBottom: 8,
		},
	},
	amountToNextRankContainer: {
		textAlign: 'end',
		[theme.breakpoints.down('xs')]: {
			textAlign: 'center',
		},
	},
	headerValueText: {
		marginLeft: theme.spacing(1),
	},
}));

export const LeaderboardAccountInformation = observer(
	(): JSX.Element => {
		const { boostOptimizer, user, router, wallet } = useContext(StoreContext);
		const connectWallet = useConnectWallet();
		const classes = useStyles();

		const nativeHoldings = boostOptimizer.nativeHoldings;
		const nonNativeHoldings = boostOptimizer.nonNativeHoldings;

		const boost = user.accountDetails?.boost;
		const rank = user.accountDetails?.boostRank;

		const currentBadgerLevel = boost ? getRankNumberFromBoost(boost) : undefined;
		const nextBadgerLevel =
			currentBadgerLevel !== undefined ? LEADERBOARD_RANKS[currentBadgerLevel - 1] : undefined;

		let amountToReachNextRank;

		if (nextBadgerLevel && nativeHoldings && nonNativeHoldings) {
			amountToReachNextRank = boostOptimizer.calculateNativeToMatchBoost(
				nativeHoldings,
				nonNativeHoldings,
				nextBadgerLevel.boostRangeStart,
			);
		}

		// Show N/A when wallet is not connected otherwise show loading skeleton
		const infoPlaceholder = !wallet.connectedAddress ? 'N/A' : <Skeleton width={30} />;

		return (
			<Grid container>
				<Grid item container alignItems="center" xs={12} sm className={classes.boostContainer}>
					<Grid container item xs={12} alignItems="center" className={classes.justifyCenterOnMobile}>
						<Typography variant="h2" display="inline">
							Boost:
						</Typography>
						<Typography variant="h2" display="inline" className={classes.headerValueText}>
							{boost !== undefined ? boost : infoPlaceholder}
						</Typography>
					</Grid>
					<Grid container item xs={12} alignItems="center" className={classes.justifyCenterOnMobile}>
						<Typography variant="h6" color="textSecondary" display="inline">
							Rank:
						</Typography>
						<Typography
							variant="h6"
							color="textSecondary"
							display="inline"
							className={classes.headerValueText}
						>
							{rank !== undefined ? rank : infoPlaceholder}
						</Typography>
					</Grid>
				</Grid>
				<Grid item container xs={12} sm className={classes.amountToNextRankContainer} alignItems="center">
					<Grid item xs={12} className={classes.calculateBoostContainer}>
						{wallet.connectedAddress ? (
							<Button
								color="primary"
								variant="contained"
								onClick={() => router.goTo(routes.boostOptimizer)}
							>
								Calculate Boost
							</Button>
						) : (
							<Button color="primary" variant="contained" onClick={connectWallet}>
								Connect Wallet
							</Button>
						)}
					</Grid>
					{nonNativeHoldings && nonNativeHoldings.eq(0) && (
						<Grid item xs={12}>
							<Typography className={classes.amountToNextRank} color="textSecondary">
								Add Non Native assets to be able to improve your boost
							</Typography>
						</Grid>
					)}
					{amountToReachNextRank && amountToReachNextRank.gt(0) && (
						<Grid item xs={12}>
							<Typography className={classes.amountToNextRank} color="textSecondary">
								<Typography className={classes.amountToNextRank} color="textPrimary" component="span">
									{`$${numberWithCommas(formatWithoutExtraZeros(amountToReachNextRank, 3))}`}
								</Typography>
								{` more `}
								<Typography className={classes.amountToNextRank} color="textPrimary" component="span">
									NATIVE
								</Typography>{' '}
								to next rank
							</Typography>
						</Grid>
					)}
				</Grid>
			</Grid>
		);
	},
);
