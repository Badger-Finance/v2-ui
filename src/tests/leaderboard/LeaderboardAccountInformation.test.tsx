import LeaderboardAccountInformation from 'components-v2/leaderboard/LeaderboardAccountInformation';
import store from 'mobx/store';
import React from 'react';
import { TEST_ADDRESS, checkSnapshot } from 'tests/utils/snapshots';

describe('LeaderboardAccountInformation', () => {
	describe('No user connected', () => {
		it('Displays fields as N/A', () => checkSnapshot(<LeaderboardAccountInformation />));
	});

	describe('User connected, no data loaded', () => {
		it('Displays fields as loading skeletons', () => {
			store.wallet.connectedAddress = TEST_ADDRESS;
			checkSnapshot(<LeaderboardAccountInformation />);
		});
	});

	describe('User connected, data loaded', () => {
		it('Displays user rank, and boost', () => {
			store.wallet.connectedAddress = TEST_ADDRESS;
			store.user.accountDetails = {
				id: TEST_ADDRESS,
				boost: 2,
				boostRank: 10,
				nativeBalance: 0,
				nonNativeBalance: 10,
				multipliers: {},
				depositLimits: {},
				value: 0,
				earnedValue: 0,
				balances: [],
			};
			checkSnapshot(<LeaderboardAccountInformation />);
		});
	});
});
