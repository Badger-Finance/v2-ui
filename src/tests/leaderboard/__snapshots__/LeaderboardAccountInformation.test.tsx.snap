// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardAccountInformation No user connected Displays fields as N/A 1`] = `
<div>
  <div
    class="MuiGrid-root-5 MuiGrid-container-6"
  >
    <div
      class="MuiGrid-root-5 makeStyles-boostContainer-2 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51 MuiGrid-grid-sm-true-53"
    >
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h2
          class="MuiTypography-root-108 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          Boost:
        </h2>
        <h2
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          N/A
        </h2>
      </div>
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h6
          class="MuiTypography-root-108 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          Rank:
        </h6>
        <h6
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          N/A
        </h6>
      </div>
    </div>
    <div
      class="makeStyles-infoContainer-4"
    >
      <button
        class="MuiButtonBase-root-168 MuiButton-root-139 MuiButton-contained-147 MuiButton-containedPrimary-148"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-140"
        >
          Connect Wallet
        </span>
        <span
          class="MuiTouchRipple-root-175"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`LeaderboardAccountInformation User connected, data loaded Displays user rank, and boost 1`] = `
<div>
  <div
    class="MuiGrid-root-5 MuiGrid-container-6"
  >
    <div
      class="MuiGrid-root-5 makeStyles-boostContainer-2 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51 MuiGrid-grid-sm-true-53"
    >
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h2
          class="MuiTypography-root-108 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          Boost:
        </h2>
        <h2
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          2.00
        </h2>
      </div>
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h6
          class="MuiTypography-root-108 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          Rank:
        </h6>
        <h6
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          10
        </h6>
      </div>
    </div>
    <div
      class="makeStyles-infoContainer-4"
    >
      <div
        class="makeStyles-buttonContainer-138"
      >
        <button
          class="MuiButtonBase-root-168 MuiButton-root-139 MuiButton-contained-147 MuiButton-containedPrimary-148"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-140"
          >
            Calculate Boost
          </span>
          <span
            class="MuiTouchRipple-root-175"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`LeaderboardAccountInformation User connected, no data loaded Displays fields as loading skeletons 1`] = `
<div>
  <div
    class="MuiGrid-root-5 MuiGrid-container-6"
  >
    <div
      class="MuiGrid-root-5 makeStyles-boostContainer-2 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51 MuiGrid-grid-sm-true-53"
    >
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h2
          class="MuiTypography-root-108 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          Boost:
        </h2>
        <h2
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h2-114 MuiTypography-displayInline-136"
        >
          <span
            class="MuiSkeleton-root-138 MuiSkeleton-text-139 MuiSkeleton-pulse-142"
            style="width: 30px;"
          />
        </h2>
      </div>
      <div
        class="MuiGrid-root-5 makeStyles-justifyCenterOnMobile-1 MuiGrid-container-6 MuiGrid-item-7 MuiGrid-align-items-xs-center-14 MuiGrid-grid-xs-12-51"
      >
        <h6
          class="MuiTypography-root-108 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          Rank:
        </h6>
        <h6
          class="MuiTypography-root-108 makeStyles-headerValueText-3 MuiTypography-h6-118 MuiTypography-colorTextSecondary-134 MuiTypography-displayInline-136"
        >
          <span
            class="MuiSkeleton-root-138 MuiSkeleton-text-139 MuiSkeleton-pulse-142"
            style="width: 30px;"
          />
        </h6>
      </div>
    </div>
    <div
      class="makeStyles-infoContainer-4"
    >
      <div
        class="makeStyles-buttonContainer-149"
      >
        <button
          class="MuiButtonBase-root-179 MuiButton-root-150 MuiButton-contained-158 MuiButton-containedPrimary-159"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-151"
          >
            Calculate Boost
          </span>
          <span
            class="MuiTouchRipple-root-186"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
